onnxruntime/test/shared_lib/custom_op_utils.cc

```diff
@@ -31,6 +36,9 @@ void MyCustomKernel::Compute(OrtKernelContext* context) {
 
   // Do computation
 #ifdef USE_CUDA
+  ASSERT_EQ(x_mem_type, OrtMemType::OrtMemTypeDefault);
```

Comment:
Prefer if the test has structured a bit differently.

We can just copy MyCustomOp/MyCustomKernel over into another variant (maybe called MyCustomOpSecondInputOnCpu/MyCustomKernelSecondInputOnCpu) and make the second input's memtype as `OrtMemTypeCPUInput` for that custom op alone. In the kernel's compute, everything remains the same as MyCustomKernel except that there is a cudaMemcpyAsync(...) before launching cuda_add(...).

This way, it tests the passing of inputs on cpu to a cuda custom op (which is basically what this feature is all about).

Reply:
Agree. Would be nice to have an example that actually demonstrates this use case.